src/main/java/controllers/BookstoreRestController.java

```diff
@@ -0,0 +1,202 @@
+        Bookstore bookstore = bookstoreRepository.findById(bookstoreID);
+        if (bookstore == null)
+            return null;
+        Book book = new Book(name, isbn, picture, description, author, publisher);
+        bookstore.addBook(book);
+        book.setBookstore(bookstore);
+        bookRepository.save(book);
+        bookstoreRepository.save(bookstore);
+        return book;
+    }
+
+
+    //Customer REST endpoints
+    @GetMapping("/api/getCustomer")
+    public Customer getCustomer(@RequestParam(value = "id") long id) {
+        return customerRepository.findById(id);
+    }
+
+    @GetMapping("/api/getCustomers")
+    public Iterable<Customer> getCustomers() {
+        return customerRepository.findAll();
+    }
+
+    @PostMapping("/api/newCustomer")
+    public Customer newCustomer(@RequestParam(value = "name") String name, @RequestParam(value = "address") String address, @RequestParam(value = "email") String email, @RequestParam(value = "phoneNumber") String phoneNumber) {
+        Customer customer = new Customer(name, address, email, phoneNumber);
+        customerRepository.save(customer);
+        return customer;
+    }
+
+
+    //Shopping Cart REST endpoints
+    @GetMapping("/api/getShoppingCart")
+    public ShoppingCart getShoppingCart(@RequestParam(value = "id") long id) {
+        return shoppingCartRepository.findById(id);
+    }
+
+    @GetMapping("/api/getShoppingCarts")
+    public Iterable<ShoppingCart> getShoppingCarts() {
+        return shoppingCartRepository.findAll();
+    }
+
+    @GetMapping("/api/getShoppingCartsByCustomer")
+    public Iterable<ShoppingCart> getShoppingCartsByCustomer(@RequestParam(value = "customerID") long customerID) {
+        Customer customer = customerRepository.findById(customerID);
+        if (customer == null)
+            return null;
+        return shoppingCartRepository.findByCustomer(customer);
+    }
+
+    @PostMapping("/api/newShoppingCart")
+    public ShoppingCart newShoppingCart(@RequestParam(value = "customerID") long customerID) {
+        Customer customer = customerRepository.findById(customerID);
+        if (customer == null)
+            return null;
+        ShoppingCart shoppingCart = new ShoppingCart(customer);
+        shoppingCartRepository.save(shoppingCart);
+        customer.setShoppingCart(shoppingCart);
+        customerRepository.save(customer);
+        return shoppingCart;
+    }
+
+    @PostMapping("/api/addBookToShoppingCart")
+    public ShoppingCart addBookToShoppingCart(@RequestParam(value = "shoppingCartID") long shoppingCartID, @RequestParam(value = "bookID") long bookID) {
+        ShoppingCart shoppingCart = shoppingCartRepository.findById(shoppingCartID);
```

Comment:
I like the idea instead of the shoppingCartId to use the customerId in order to get the cart from the customer object. Unless you think it is better to pass the shoppingCartId explicitly? I feel for most accesses to a customer's attributes, the customerId should be used to access them, it would provide a more consistent API across our endpoints. But this of course is debatable. If we change it to that, the name could be something like 'addBookToCustomersShoppingCart'

Reply:
I think that makes a lot more sense.
Addressed in 0f27f09dc1598aeea21d1460bc54bc99cf5a5a3f